src/mpid/ch4/netmod/ofi/ofi_spawn.c

```diff
@@ -6,10 +6,17 @@
 #include "mpidimpl.h"
 #include "ofi_impl.h"
 #include "ofi_noinline.h"
+#include "uthash.h"
 
 /* NOTE: all these functions assume the caller to enter VCI-0 critical section */
 
+/* NOTE: MPIDI_OFI_dynamic_{send,recv} is an "any source" message that is used for initial MPI_Comm_{connect/accept},
+ *       while MPIDI_OFI_dynamic_sendrecv require exact source matching. That latter uses get_dynamic_match_bits
+ *       for match_bits.
```

Comment:
maybe this comment is from a previous version where both cases didn't use `get_dynamic_match_bits`?

Reply:
is there still a danger of an "any source" recv jumping in front of the recv with the source and still causing issues?

Reply:
I added the last commit to separate the matching space between the dynamic connection (send and recv) from the dynamic data exchange (MPIDI_NM_dynamic_sendrecv).